app/models/project.rb

```diff
@@ -315,6 +318,50 @@ def orphan_each_draft(drafts)
     end
   end
 
+  ##############################################################################
+  #
+  #  :section: Dates
+  #
+  ##############################################################################
+
+  def current?
+    !future? && !past?
+  end
+
+  def dates_exclude?(date)
+    !dates_include?(date)
+  end
+
+  def dates_include?(date)
+    starts_no_later_than?(date) && ends_no_earlier_than?(date)
+  end
+
```

Comment:
Should I try to turn these into ARELized, scoped, AR queries.

Reply:
Not a bad idea, and they're pretty easy to write.